File: java/resistor-color-duo/src/main/java/ResistorColorDuo.java
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

class ResistorColorDuo {
    private final String[] COLORS = new String[] {
            "black",
            "brown",
            "red",
            "orange",
            "yellow",
            "green",
            "blue",
            "violet",
            "grey",
            "white"
    };

    int value(String[] colors) {
        List<String> intList = Arrays.stream(colors)
                .map(color -> String.valueOf(Arrays.asList(COLORS)
                        .indexOf(color))).collect(Collectors.toList());

        return Integer.parseInt(String.join("", intList)
                .substring(0, 2));
    }
}

File: go/resistor-color-duo/resistor_color_duo.go
package resistorcolorduo

import "strconv"

var resistorColorMap = map[string]rune{
	"black":  '0',
	"brown":  '1',
	"red":    '2',
	"orange": '3',
	"yellow": '4',
	"green":  '5',
	"blue":   '6',
	"violet": '7',
	"grey":   '8',
	"white":  '9',
}

// Value should return the resistance value of a resistor with a given colors.
func Value(colors []string) int {
	v, _ := strconv.Atoi(
		string([]rune{resistorColorMap[colors[0]], resistorColorMap[colors[1]]}),
	)
	return v
}
